### PR TITLE
GAC Documentation

### PR DESCRIPTION
**Pull request by hunterbarber**
## Fixes/Resolves:
n/a

## Summary/Motivation:
First addition of documentation for the GAC unit model

## Changes proposed in this PR:
n/a

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/technical_reference/unit_models/gac.rst [hunk 1]
```diff
@@ -0,0 +1,248 @@
+Granular Activated Carbon (GAC)
+====================
```
Comment (adam-a-a) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
Granular Activated Carbon (GAC)
===============================
```

File: docs/technical_reference/unit_models/gac.rst [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+   * assumes isothermal conditions
+   * model performance is independent of a gravity-fed or pressurized GAC unit, therefore assumes isobaric
+
+.. index::
+   pair: watertap.unit_models.gac;gac
+
+.. currentmodule:: watertap.unit_models.gac
+
+Introduction
+-----------
```
Comment (adam-a-a) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
Introduction
------------
```

File: docs/technical_reference/unit_models/gac.rst [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+   "Throughput equation parameter 4", ":math:`b_4`", "b4", "None", ":math:`\text{dimensionless}`"
+
+The following variables are only built when specific configuration options are selected.
+
+if ``film_transfer_coefficient_type`` or ``surface_diffusion_coefficient_type`` is set to ``calculated``:
+
+.. csv-table::
+   :header: "Description", "Symbol", "Variable Name", "Index", "Units"
+
+   "Molecular diffusion coefficient", ":math:`D_l`", "diffus_liq", "None", ":math:`\text{m}^2\text{\s}`"
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
   "Molecular diffusion coefficient", ":math:`D_l`", "diffus_liq", "None", ":math:`\text{m}^2\text{/s}`"
```

**Comment by hunterbarber** [comment]
Uploaded PR to review .rst conversion into the readthedocs
I'm doing an ongoing review of all the formatting but will respond to this when it looks cleaner

**Comment by hunterbarber** [comment]
Should be formatted enough for review at this point

File: docs/technical_reference/unit_models/gac.rst [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+Introduction
+------------
+The implemented model for estimating GAC performance is adapted from a simplified model originally presented in Hand, 1984 and
+further elaborated in Crittenden, 2012. This formulation is denoted as the constant pattern homogeneous surface diffusion model (CPHSDM).
+As a GAC system is operated as a batch process, a mass transfer zone (MTZ) is formed in the bed where a concentration profile, or
+breakthrough curve, developeds as a function of the adsorption properties. This MTZ is bounded by saturated GAC upstream and
+fresh GAC downstream. The CPHSDM is valid under the assumption that the shape of the MTZ is constant as it travels through the
+bed and a constant pattern solution (CPS) may be determined. The CPS is calculated through a multistep procedure utilizing common
+dimensionless groups applied in polynomial fits to determine performance. Therefore, coefficients used in the polynomial must be
+derived from experimental dataof the intended system to produce valid results. Coefficients for common compounds treated by GAC may be
```
Comment (luohezhiming) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
derived from experimental data of the intended system to produce valid results. Coefficients for common compounds treated by GAC may be
```

**Review by luohezhiming — CHANGES_REQUESTED** [review]
LGTM. I only find one small typo.

**Review by kurbansitterley — APPROVED** [review]
LGTM. I am a big fan of putting anything that is referencing something in the code in `monospaced font` (like the name of the Python variable) for aesthetic reasons. But I see that isn't done in any other model documentation.

**Review by luohezhiming — APPROVED** [review]

File: docs/technical_reference/unit_models/gac.rst [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+   "Mass of fresh GAC in the bed", ":math:`M_{GAC}`", "bed_mass_gac", "None", ":math:`\text{kg}`"
+   "Superficial velocity", ":math:`v_s`", "velocity_sup", "None", ":math:`\text{m/s}`"
+   "Interstitial velocity", ":math:`v_i`", "velocity_int", "None", ":math:`\text{m/s}`"
+   "GAC particle porosity", ":math:`\epsilon_p`", "particle_porosity", "None", ":math:`\text{dimensionless}`"
+   "GAC particle apparent density", ":math:`\rho_a`", "particle_dens_app", "None", ":math:`\text{kg/}\text{m}^3`"
+   "GAC particle bulk density", ":math:`\rho_b`", "particle_dens_bulk", "None", ":math:`\text{kg/}\text{m}^3`"
+   "GAC particle solid density", ":math:`\rho_s`", "particle_dens_sol", "None", ":math:`\text{kg/}\text{m}^3`"
+   "GAC particle diameter", ":math:`d_p`", "particle_dia", "None", ":math:`\text{m}`"
+   "Average dimensionless concentration of the effluent in the operating duration", ":math:`\frac{\bar{C}}{C_{0}}\bigg{|}_{z=L}`", "conc_ratio_avg", "None", ":math:`\text{dimensionless}`"
+   "Dimensionless concentration of the effluent at the time of replacement", ":math:`\frac{C}{C_{0}}\bigg{|}_{z=L}`", "conc_ratio_avg", "None", ":math:`\text{dimensionless}`"
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
   "Dimensionless concentration of the effluent at the time of replacement", ":math:`\frac{C}{C_{0}}\bigg{|}_{z=L,t=t_{op}}`", "conc_ratio_replace", "None", ":math:`\text{dimensionless}`"
```

File: docs/technical_reference/unit_models/gac.rst [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+   "Mass of fresh GAC in the bed", ":math:`M_{GAC}`", "bed_mass_gac", "None", ":math:`\text{kg}`"
+   "Superficial velocity", ":math:`v_s`", "velocity_sup", "None", ":math:`\text{m/s}`"
+   "Interstitial velocity", ":math:`v_i`", "velocity_int", "None", ":math:`\text{m/s}`"
+   "GAC particle porosity", ":math:`\epsilon_p`", "particle_porosity", "None", ":math:`\text{dimensionless}`"
+   "GAC particle apparent density", ":math:`\rho_a`", "particle_dens_app", "None", ":math:`\text{kg/}\text{m}^3`"
+   "GAC particle bulk density", ":math:`\rho_b`", "particle_dens_bulk", "None", ":math:`\text{kg/}\text{m}^3`"
+   "GAC particle solid density", ":math:`\rho_s`", "particle_dens_sol", "None", ":math:`\text{kg/}\text{m}^3`"
+   "GAC particle diameter", ":math:`d_p`", "particle_dia", "None", ":math:`\text{m}`"
+   "Average dimensionless concentration of the effluent in the operating duration", ":math:`\frac{\bar{C}}{C_{0}}\bigg{|}_{z=L}`", "conc_ratio_avg", "None", ":math:`\text{dimensionless}`"
+   "Dimensionless concentration of the effluent at the time of replacement", ":math:`\frac{C}{C_{0}}\bigg{|}_{z=L,t=t_{op}}`", "conc_ratio_replace", "None", ":math:`\text{dimensionless}`"
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
   "Dimensionless concentration of the effluent at the time of replacement", ":math:`\frac{C}{C_{0}}\bigg{|}_{z=L,\,t=t_{op}}`", "conc_ratio_replace", "None", ":math:`\text{dimensionless}`"
```

File: docs/technical_reference/unit_models/gac.rst [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+    to a specific mass of contaminant adsorbed during the operation time and is used to transform the process to a steady state approximation.
+
+Degrees of Freedom
+------------------
+In the default configuration of the GAC unit model there are 18 degrees of freedom in addition to the inlet state variables
+(i.e. temperature, pressure, component flowrates) that should be fixed for the model to be fully specified.
+In association with using the Freundlich adsorption isotherm and empirical model, the following 9 variables are almost always fixed
+and may be derived from experimental data:
+
+   * Freundlich isotherm ":math:`k`" parameter
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
   * Freundlich isotherm :math:`k` parameter
```

File: docs/technical_reference/unit_models/gac.rst [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+   * Freundlich isotherm ":math:`\frac{1}{n}`" parameter
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
   * Freundlich isotherm :math:`\frac{1}{n}` parameter
```

File: docs/technical_reference/unit_models/gac.rst [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+   "GAC particle diameter", ":math:`d_p`", "particle_dia", "None", ":math:`\text{m}`"
+   "Average dimensionless concentration of the effluent in the operating duration", ":math:`\frac{\bar{C}}{C_{0}}\bigg{|}_{z=L}`", "conc_ratio_avg", "None", ":math:`\text{dimensionless}`"
+   "Dimensionless concentration of the effluent at the time of replacement", ":math:`\frac{C}{C_{0}}\bigg{|}_{z=L,\,t=t_{op}}`", "conc_ratio_replace", "None", ":math:`\text{dimensionless}`"
+   "Approximate saturation of the GAC in the bed at the time of replacement", ":math:`\frac{\bar{q}}{q_{e}}\bigg{|}_{t=t_{op}}`", "gac_saturation_replace", "None", ":math:`\text{dimensionless}`"
+   "Empty bed contact time", ":math:`EBCT`", "ebct", "None", ":math:`\text{s}`"
+   "Mass throughput ratio", ":math:`T`", "mass_throughput", "None", ":math:`\text{dimensionless}`"
+   "Residence time", ":math:`\tau`", "res_time", "None", ":math:`\text{s}`"
+   "Elapsed operation time", ":math:`t_{op}`", "elap_time", "None", ":math:`\text{s}`"
+   "Steady state GAC replacement rate", ":math:`\dot{m}_{GAC}`", "gac_mass_replace_rate", "None", ":math:`\text{s}`"
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
   "Steady state GAC replacement rate", ":math:`\dot{m}_{GAC}`", "gac_mass_replace_rate", "None", ":math:`\text{m/s}`"
```

File: docs/technical_reference/unit_models/gac.rst [hunk 10]
```diff
@@ -0,0 +1,251 @@
+Granular Activated Carbon (GAC)
+===============================
+.. note::
+
+    The GAC model is an ongoing project and subject to change, as is this documentation.
+
+This is a empirical-performance-based granular activated carbon (GAC) model that works under the following criteria and assumptions
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
This is an empirical-performance-based granular activated carbon (GAC) model that works under the following criteria and assumptions:
```

File: docs/technical_reference/unit_models/gac.rst [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+   * supports a single liquid phase only
+   * supports a single solute and single solvent (water) only
+   * supports steady-state only
+   * assumes isothermal conditions
+   * model performance is independent of a gravity-fed or pressurized GAC unit, therefore assumes isobaric
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
   * model performance is independent of a gravity-fed or pressurized GAC unit, therefore assumes isobaric conditions
```

File: docs/technical_reference/unit_models/gac.rst [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+further elaborated in Crittenden, 2012. This formulation is denoted as the constant pattern homogeneous surface diffusion model (CPHSDM).
+As a GAC system is operated as a batch process, a mass transfer zone (MTZ) is formed in the bed where a concentration profile, or
+breakthrough curve, developeds as a function of the adsorption properties. This MTZ is bounded by saturated GAC upstream and
+fresh GAC downstream. The CPHSDM is valid under the assumption that the shape of the MTZ is constant as it travels through the
+bed and a constant pattern solution (CPS) may be determined. The CPS is calculated through a multistep procedure utilizing common
+dimensionless groups applied in polynomial fits to determine performance. Therefore, coefficients used in the polynomial must be
+derived from experimental data of the intended system to produce valid results. Coefficients for common compounds treated by GAC may be
+found in both Hand, 1984 and Crittenden, 2012. The model is estimated to have within 10% error and therefore may be applied to bed lengths
+shorter than the minimum length determined by the CPHSDM within the error threshold (or lengths greater than the minimum).
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
shorter than the minimum length determined by the CPHSDM within the error threshold (in addition to being applicable to bed lengths greater than the minimum length determined by the CPHSDM).
```

File: docs/technical_reference/unit_models/gac.rst [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+    shown as a function of the elapsed time in operation of the bed. The dashed line indicates a breakthrough time where the bed operation will
+    be stopped (and the partially saturated GAC will be replaced) when the effluent concentration is 50% of the inlet. (c) For the breakthrough
+    time in b, some of the MTZ is still contained within the bed and therefore the GAC particles in this zone are not fully saturated. The
+    trapezoid rule for integration is used to approximate the degree of saturation of the entire bed. This degree of saturation corresponds
+    to a specific mass of contaminant adsorbed during the operation time and is used to transform the process to a steady state approximation.
+
+Degrees of Freedom
+------------------
+In the default configuration of the GAC unit model there are 18 degrees of freedom in addition to the inlet state variables
+(i.e. temperature, pressure, component flowrates) that should be fixed for the model to be fully specified.
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
(i.e., temperature, pressure, component flowrates) that should be fixed for the model to be fully specified.
```

File: docs/technical_reference/unit_models/gac.rst [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+   * molal volume of the solute
+   * GAC particle sphericity
+   * tortuosity of the path that the adsorbate must take as compared to the radius
+   * surface-to-pore diffusion flux ratio
+
+Model Structure
+------------------
+The GAC model consists of 1 ControlVolume0DBlock (process_flow) for the process flow of the water treatment train.
+The process flow includes 2 StateBlocks (inlet and outlet) which are used for mass and momentum balances.
+It also includes 1 StateBlock (adsorbed) for the solute that is adsorbed into the GAC parties.
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
It also includes 1 StateBlock (adsorbed) for the solute that is adsorbed into the GAC particles.
```

File: docs/technical_reference/unit_models/gac.rst [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+Variables
+----------
+
+.. csv-table::
+   :header: "Description", "Symbol", "Variable Name", "Index", "Units"
+
+   "Freundlich isotherm k parameter", ":math:`k`", "freund_k", "None", ":math:`\left(\text{m}^3\text{/kg}\right)^\left( \frac{1}{n} \right)`"
+   "Freundlich isotherm 1/n parameter", ":math:`\frac{1}{n}`", "freund_ninv", "None", ":math:`\text{dimensionless}`"
+   "Equilibrium concentration of adsorbed phase with liquid phase", ":math:`q_e`", "equil_conc", "None", ":math:`\left( \text{kg}_\text{adsorbate}\text{/kg}_\text{adsorbent} \right)`"
+   "Mass of contaminant absorbed at the time of replacement", ":math:`M_{solute}`", "mass_adsorbed", "None", ":math:`\text{kg}`"
```
Comment (hunterbarber) [inline docs/technical_reference/unit_models/gac.rst]:
```suggestion
   "Mass of contaminant adsorbed at the time of replacement", ":math:`M_{solute}`", "mass_adsorbed", "None", ":math:`\text{kg}`"
```

**Pull request merged**